client/src/modules/journal/modals/search.modal.js

```diff
@@ -80,8 +58,18 @@ function JournalSearchModalController(Instance, Notify,
   // custom filter account_id - assign the value to the searchQueries object
   vm.onSelectAccount = function onSelectAccount(account) {
     vm.searchQueries.account_id = account.id;
+    displayValues.account_id = String(account.number).concat(' - ', account.label);
```

Comment:
There is no need to use `String` - what you are looking for is:
```js
displayValues.account_id = account.hrLabel;
```

The `AccountService` formats labels after they are downloaded - see [this line](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/accounts/accounts.service.js#L45).